File: app/src/main/java/ggeorgip/androidtemplate/di/modules/RepositoryModule.java
package ggeorgip.androidtemplate.di.modules;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import ggeorgip.androidtemplate.repositories.IMainRepository;
import ggeorgip.androidtemplate.repositories.MainRepository;

/**
 * Created by Georgi Petrov on 7.8.2018 г..
 */
@Module
public class RepositoryModule {

    @Provides
    @Singleton
    public IMainRepository provideMainRepository() {
        return new MainRepository();
    }

}

File: app/src/main/java/ggeorgip/androidtemplate/di/modules/RepositoryModule.kt
package ggeorgip.androidtemplate.di.modules

import dagger.Module
import dagger.Provides
import ggeorgip.androidtemplate.repositories.IMainRepository
import ggeorgip.androidtemplate.repositories.MainRepository
import javax.inject.Singleton

/**
 * Created by Georgi Petrov on 7.8.2018 г..
 */
@Module
class RepositoryModule {

    @Provides
    @Singleton
    fun provideMainRepository(): IMainRepository {
        return MainRepository()
    }

}
